applications/GeoMechanicsApplication/tests/cpp_tests/test_generalized_newmark_T_scheme.cpp

```diff
@@ -0,0 +1,180 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_strategies/schemes/generalized_newmark_T_scheme.hpp"
+#include "spaces/ublas_space.h"
+#include "testing/testing.h"
+
+using namespace Kratos;
+using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+
+namespace Kratos::Testing {
+
+GeneralizedNewmarkTScheme<SparseSpaceType, LocalSpaceType> CreateValidScheme()
+{
+    GeneralizedNewmarkTScheme<SparseSpaceType, LocalSpaceType> result(0.75);
+    return result;
+}
+
+ModelPart& CreateValidModelPart(Model& rModel)
+{
+    auto& result = rModel.CreateModelPart("dummy", 2);
+    result.AddNodalSolutionStepVariable(TEMPERATURE);
+    result.AddNodalSolutionStepVariable(DT_TEMPERATURE);
+    auto p_node = result.CreateNewNode(0, 0.0, 0.0, 0.0);
+    p_node->AddDof(TEMPERATURE);
+
+    return result;
+}
+
+
+KRATOS_TEST_CASE_IN_SUITE(CheckBackwardEulerQuasistaticTScheme_WithAllNecessaryParts_Returns0,
+                          KratosGeoMechanicsFastSuite)
+{
+    Model model;
+    auto scheme = CreateValidScheme();
+    const auto& model_part = CreateValidModelPart(model);
+
+    KRATOS_EXPECT_EQ(scheme.Check(model_part), 0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForInvalidTheta_CheckBackwardEulerQuasistaticTScheme_Throws,
+                          KratosGeoMechanicsFastSuite)
+{
+    constexpr int invalid_theta = -2;
```

Comment:
Since `theta` is a `double`, let's declare it as such:
```suggestion
    constexpr auto invalid_theta = -2.0;
```

Reply:
Whoops, good catch, changed!